dgraph/cmd/zero/http.go

```diff
@@ -99,11 +99,17 @@ func (st *state) assign(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	m := jsonpb.Marshaler{EmitDefaults: true}
-	if err := m.Marshal(w, ids); err != nil {
+	m := protojson.MarshalOptions{EmitUnpopulated: true}
+	buf, err := m.Marshal(ids)
```

Comment:
can we use MarshalAppend here instead?

Reply:
We can use MarshalAppend, but it does not offer any benefit; it will work the same as Marshal.

Reply:
the benefit would be to be able to write to `w` directly instead of writing to buffer first.